sled-agent/src/sled_agent.rs

```diff
@@ -691,6 +697,11 @@ impl SledAgent {
         .unwrap(); // we retry forever, so this can't fail
     }
 
+    /// Accesses the [SupportBundleManager] API.
+    pub(crate) fn as_support_bundle_storage(&self) -> SupportBundleManager<'_> {
+        SupportBundleManager::new(&self.log, self.storage())
+    }
```

Comment:
Rather than directly implementing the support bundle API as part of the Sled Agent, I split it into a different structure to make testing easier.

That's why I'm calling this function in the `http_entrypoints`